scripts/testing/database-emulator-test.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const { spawn } = require('child-process-promise');
+
+const EMULATOR_LINK =
+  'https://firebase.google.com/docs/database/downloads/database-emulator.jar';
+const EMULATOR_DIRECTORY = path.join(os.tmpdir(), 'firebase-emulators');
+const EMULATOR_FILENAME = 'database-emulator.jar';
+
+const EMULATOR_PORT = 8088;
+const EMULATOR_NAMESPACE = 'test-emulator';
+
+async function delay(seconds) {
+  return new Promise(resolve => setTimeout(resolve, 1000 * seconds));
+}
+
+async function downloadEmulator() {
+  return new Promise((resolve, reject) => {
+    console.log(`Creating directory [${EMULATOR_DIRECTORY}] ...`);
+    fs.mkdir(EMULATOR_DIRECTORY, err => {
+      if (err && err.code !== 'EEXIST') reject(err);
+      console.log(`Downloading emulator binary from [${EMULATOR_LINK}] ...`);
+      // const promise = spawn('firebase', ['setup:emulators:database']);
+      const promise = spawn('curl', [EMULATOR_LINK, '-o', EMULATOR_FILENAME], {
+        cwd: EMULATOR_DIRECTORY,
+        stdio: 'inherit'
+      });
+      resolve(promise);
+    });
+  });
+}
+
+async function launchEmulator(locked) {
```

Comment:
It's not obvious to me what `locked` indicates.  Please rename and/or add comments.

Reply:
Done.